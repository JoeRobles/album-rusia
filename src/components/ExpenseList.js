import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import ExpensesSummary from './ExpensesSummary';

export const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    <table>
      <thead>
      <tr>
        <th>Description</th>
        <th>Amount</th>
        <th>Created at</th>
      </tr>
      </thead>
      <tbody>
      {
        props.expenses.length === 0 ? (
          <tr><td>No expenses</td></tr>
        ) : (
          props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} {...props} />;
          })
        )
      }
      </tbody>
      <ExpensesSummary/>
    </table>
  </div>
);

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseList);
