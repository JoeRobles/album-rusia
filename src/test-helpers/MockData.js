import moment from 'moment';

export const expensesReducerDefaultState = [
  {
    id: 'poijasdfhwer',
    description: 'January Rent',
    note: 'This was the final payment for that address',
    amount: 54500,
    createdAt: 1514954400000,
  },
  {
    id: 'lmdfmlgkdnfg',
    description: 'water bill',
    amount: 100,
    createdAt: 1515085200000,
  },
  {
    id: 'fhgerthhrdhd',
    description: 'gas bill',
    amount: 300,
    createdAt: 1515140400000,
  },
  {
    id: 's√±ojglsjslkg',
    description: 'internet bill',
    amount: 200,
    createdAt: 1515267500000,
  },
];

export const filtersReducerDefaultState = {
  text: '',
  sortBy: 'date',
  asc: true,
  startDate: moment().startOf('month'),
  endDate: moment().endOf('month'),
};