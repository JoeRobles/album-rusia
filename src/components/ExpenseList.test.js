import React from 'react';
import { shallow } from 'enzyme';
import { expensesReducerDefaultState } from '../test-helpers/MockData';
import { ExpenseList } from './ExpenseList';

it('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expensesReducerDefaultState} />);
  expect(wrapper).toMatchSnapshot();
});

it('should render ExpenseList with empty message', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
