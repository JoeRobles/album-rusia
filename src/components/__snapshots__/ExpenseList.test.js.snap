// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Amount
        </th>
        <th>
          Created at
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          No expenses
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Amount
        </th>
        <th>
          Created at
        </th>
      </tr>
    </thead>
    <tbody>
      <Connect(Component)
        amount={54500}
        createdAt={1514954400000}
        description="January Rent"
        expenses={
          Array [
            Object {
              "amount": 54500,
              "createdAt": 1514954400000,
              "description": "January Rent",
              "id": "poijasdfhwer",
              "note": "This was the final payment for that address",
            },
            Object {
              "amount": 100,
              "createdAt": 1515085200000,
              "description": "water bill",
              "id": "lmdfmlgkdnfg",
            },
            Object {
              "amount": 300,
              "createdAt": 1515140400000,
              "description": "gas bill",
              "id": "fhgerthhrdhd",
            },
            Object {
              "amount": 200,
              "createdAt": 1515267500000,
              "description": "internet bill",
              "id": "sñojglsjslkg",
            },
          ]
        }
        id="poijasdfhwer"
        key="poijasdfhwer"
        note="This was the final payment for that address"
      />
      <Connect(Component)
        amount={100}
        createdAt={1515085200000}
        description="water bill"
        expenses={
          Array [
            Object {
              "amount": 54500,
              "createdAt": 1514954400000,
              "description": "January Rent",
              "id": "poijasdfhwer",
              "note": "This was the final payment for that address",
            },
            Object {
              "amount": 100,
              "createdAt": 1515085200000,
              "description": "water bill",
              "id": "lmdfmlgkdnfg",
            },
            Object {
              "amount": 300,
              "createdAt": 1515140400000,
              "description": "gas bill",
              "id": "fhgerthhrdhd",
            },
            Object {
              "amount": 200,
              "createdAt": 1515267500000,
              "description": "internet bill",
              "id": "sñojglsjslkg",
            },
          ]
        }
        id="lmdfmlgkdnfg"
        key="lmdfmlgkdnfg"
      />
      <Connect(Component)
        amount={300}
        createdAt={1515140400000}
        description="gas bill"
        expenses={
          Array [
            Object {
              "amount": 54500,
              "createdAt": 1514954400000,
              "description": "January Rent",
              "id": "poijasdfhwer",
              "note": "This was the final payment for that address",
            },
            Object {
              "amount": 100,
              "createdAt": 1515085200000,
              "description": "water bill",
              "id": "lmdfmlgkdnfg",
            },
            Object {
              "amount": 300,
              "createdAt": 1515140400000,
              "description": "gas bill",
              "id": "fhgerthhrdhd",
            },
            Object {
              "amount": 200,
              "createdAt": 1515267500000,
              "description": "internet bill",
              "id": "sñojglsjslkg",
            },
          ]
        }
        id="fhgerthhrdhd"
        key="fhgerthhrdhd"
      />
      <Connect(Component)
        amount={200}
        createdAt={1515267500000}
        description="internet bill"
        expenses={
          Array [
            Object {
              "amount": 54500,
              "createdAt": 1514954400000,
              "description": "January Rent",
              "id": "poijasdfhwer",
              "note": "This was the final payment for that address",
            },
            Object {
              "amount": 100,
              "createdAt": 1515085200000,
              "description": "water bill",
              "id": "lmdfmlgkdnfg",
            },
            Object {
              "amount": 300,
              "createdAt": 1515140400000,
              "description": "gas bill",
              "id": "fhgerthhrdhd",
            },
            Object {
              "amount": 200,
              "createdAt": 1515267500000,
              "description": "internet bill",
              "id": "sñojglsjslkg",
            },
          ]
        }
        id="sñojglsjslkg"
        key="sñojglsjslkg"
      />
    </tbody>
  </table>
</div>
`;
